lib/controller.dart

```diff
@@ -297,13 +337,53 @@ void _computeChartBuckets(List<model.Chart> charts) {
           filterValuesPercent.roundToDecimal(2),
           comparisonFilterValuesPercent.roundToDecimal(2)
         ];
+
+        // todo: handle normalisation of time series charts
       }
     }
   }
 
   logger.debug('Computed chart buckets ${charts}');
 }
 
+String _generateDateTimeKey(DateTime dateTime, model.TimeAggregate aggregate) {
+  switch (aggregate) {
+    case model.TimeAggregate.day:
+      dateTime = DateTime(dateTime.year, dateTime.month, dateTime.day, 0, 0, 0);
+      break;
+    case model.TimeAggregate.hour:
+      dateTime = DateTime(
+          dateTime.year, dateTime.month, dateTime.day, dateTime.hour, 0, 0);
+      break;
+    default:
+      logger.error('Time series chart aggregate ${aggregate} not handled');
+  }
+  return STANDARD_DATE_TIME_FORMAT.format(dateTime);
+}
+
+Map<String, num> _generateEmptyDateTimeBuckets(
+    DateTime start, DateTime end, model.TimeAggregate aggregate) {
+  var timeBucket = Map<String, num>();
+  var currentTime = DateTime(start.year, start.month, start.day, 0, 0, 0);
+  var endTime = DateTime(end.year, end.month, end.day, 23, 59, 59);
+  var durationToAdd;
+  switch (aggregate) {
+    case model.TimeAggregate.day:
+      durationToAdd = Duration(days: 1);
+      break;
+    case model.TimeAggregate.hour:
+      durationToAdd = Duration(hours: 1);
+      break;
+    default:
```

Comment:
you can skip adding the default case. that way, if you add another value to the enum, it will show you a warning in the editor that it's not being handled by the switch

Reply:
Fixed